cobigen/core-artifact-retriever/src/test/java/com/devonfw/cobigen/retriever/MavenMirrorTest.java

```diff
@@ -0,0 +1,204 @@
+package com.devonfw.cobigen.retriever;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.devonfw.cobigen.api.exception.CobiGenRuntimeException;
+import com.devonfw.cobigen.retriever.settings.MavenMirror;
+import com.devonfw.cobigen.retriever.settings.MavenSettings;
+import com.devonfw.cobigen.retriever.settings.to.model.MavenSettingsMirrorModel;
+import com.devonfw.cobigen.retriever.settings.to.model.MavenSettingsModel;
+import com.devonfw.cobigen.retriever.settings.to.model.MavenSettingsRepositoryModel;
+
+/**
+ * Test class for MavenMirror
+ */
+public class MavenMirrorTest {
+
+  /** Test data root path */
+  private static final String testdataRoot = "src/test/resources/testdata/unittest/MavenMirrorTest";
+
+  /** Repositories on which the tests are performed on */
+  private static List<MavenSettingsRepositoryModel> repositoriesList;
+
+  /** Mirrors on which the tests are performed on */
+  private static List<MavenSettingsMirrorModel> mirrorsList;
+
+  /**
+   * Used to initialize data needed for the tests
+   */
+  @BeforeClass
+  public static void setUpClass() {
+
+    MavenSettingsModel model;
+    String content;
+    try {
+      content = new String(Files.readAllBytes(Paths.get(testdataRoot).resolve("settings.xml")));
+    } catch (IOException e) {
+      throw new CobiGenRuntimeException("Unable to read test settings.xml", e);
```

Comment:
This exception should not be thrown within a test.

Reply:
Done